trade/src/main/java/bisq/trade/bisq_easy/protocol/messages/BisqEasyTakeOfferRequestHandler.java

```diff
@@ -125,4 +117,43 @@ private void commitToModel(ContractSignatureData takersContractSignatureData, Co
         trade.getTaker().getContractSignatureData().set(takersContractSignatureData);
         trade.getMaker().getContractSignatureData().set(makersContractSignatureData);
     }
-}
\ No newline at end of file
+
+    private void validateAmount(BisqEasyOffer takersOffer, BisqEasyContract takersContract, boolean isBase) {
+        double tolerancePercentage = 0.01;
+
+        Optional<Monetary> minAmount = getMinAmount(takersOffer, takersContract, isBase);
+        Optional<Monetary> maxAmount = getMaxAmount(takersOffer, takersContract, isBase);
+
+        checkArgument(minAmount.isPresent() && maxAmount.isPresent(), "No market price available for validation.");
+
+        long toleranceAmountForMin = (long) (minAmount.get().getValue() * tolerancePercentage);
+        long toleranceAmountForMax = (long) (maxAmount.get().getValue() * tolerancePercentage);
+        long minAmountWithTolerance = minAmount.get().getValue() - toleranceAmountForMin;
+        long maxAmountWithTolerance = maxAmount.get().getValue() + toleranceAmountForMax;
+
+        long amount = isBase ? takersContract.getBaseSideAmount() : takersContract.getQuoteSideAmount();
+        String errorMsg = "Market price deviation is too big.";
+        checkArgument(amount >= minAmountWithTolerance, errorMsg);
+        checkArgument(amount <= maxAmountWithTolerance, errorMsg);
+    }
+
+    private Optional<Monetary> getMinAmount(BisqEasyOffer takersOffer, BisqEasyContract takersContract, boolean isBase) {
+        return isBase
+                ? OfferAmountUtil.findBaseSideMinOrFixedAmount(
```

Comment:
I like the style for the ternary op with the ? and : in the next line, similar like in fluid interfaces the . in the next line.
If others agree we could add that to the code guide lines to unify our code style.